### PR TITLE
Adding changelog for release-0.12

### PR DESCRIPTION
**Pull request by panktishah26**
*Description of changes:*
Adding change log for the 0.12 release. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/changelog.md [hunk 1]
```diff
@@ -7,8 +7,21 @@ weight: 35
...
+### Added
+- Add support for EKS Anywhere with provider Cloudstack
+- Add support to upgrade Bare Metal cluster
+- Add support for using Registry Mirror for Bare Metal
+- RHEL availability for vSphere and Bare Metal cluster
+- Allow authenticated image pull using Registry Mirror for vSphere cluster
+- 
+
+### Fixed
+
```
Comment (pokearu) [inline docs/content/en/docs/reference/changelog.md:23]: #2814 #3391 #3560
These 3 issues were fixed. Can we add a note for that? 😄

File: docs/content/en/docs/reference/changelog.md [hunk 2]
```diff
@@ -7,8 +7,21 @@ weight: 35
...
+## [v0.12.0]
+
+### Added
+- Add support for EKS Anywhere with provider Cloudstack
+- Add support to upgrade Bare Metal cluster
+- Add support for using Registry Mirror for Bare Metal
+- RHEL availability for vSphere and Bare Metal cluster
+- Allow authenticated image pull using Registry Mirror for vSphere cluster
+-
```
Comment (vivek-koppuru) [inline docs/content/en/docs/reference/changelog.md:20]:
```suggestion
- Add option to disable vSphere CSI driver [#3148](https://github.com/aws/eks-anywhere/issues/3148)
```

File: docs/content/en/docs/reference/changelog.md (hunk 1 above)
Comment (vivek-koppuru) [inline docs/content/en/docs/reference/changelog.md:23]:
```suggestion

- Remove special cilium network policy with `policyEnforcementMode` is set to `always` due to lack of pod network connectivity for vsphere csi
```

File: docs/content/en/docs/reference/changelog.md (hunk 2 above)
Comment (abhinavmpandey08) [inline docs/content/en/docs/reference/changelog.md:20]:
```suggestion
- Add support for skipping load balancer deployment for Bare Metal so users can use their own load balancers [#3608](https://github.com/aws/eks-anywhere/pull/3608)
```

File: docs/content/en/docs/reference/changelog.md [hunk 3]
```diff
@@ -7,8 +7,21 @@ weight: 35
...
+- 
+
+## [v0.12.0]
+
+### Added
+- Add support for EKS Anywhere with provider Cloudstack
+- Add support to upgrade Bare Metal cluster
+- Add support for using Registry Mirror for Bare Metal
+- RHEL availability for vSphere and Bare Metal cluster
+- Allow authenticated image pull using Registry Mirror for vSphere cluster
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/reference/changelog.md:19]:
```suggestion
- Allow authenticated image pull using Registry Mirror for Ubuntu on vSphere cluster
```

File: docs/content/en/docs/reference/changelog.md [hunk 4]
```diff
@@ -7,8 +7,21 @@ weight: 35
 ## Unreleased
 
 ### ⚠️ Breaking changes
-- Certificates signed with SHA-1 are not supported anymore for Registry Mirror. Users with a registry mirror and providing a custom CA cert will need to rotate the certificate served by the registry mirror endpoint before using the new EKS-A version. This is true for both new clusters (`create cluster` command) and existing clusters (`upgrade cluster` command).
-- The `--source` option was removed from several package commands. Use either `--kube-version` for registry or `--cluster` for cluster.
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply docs/content/en/docs/reference/changelog.md:11]: Yes

File: docs/content/en/docs/reference/changelog.md [hunk 5]
```diff
@@ -10,6 +10,27 @@ weight: 35
 - Certificates signed with SHA-1 are not supported anymore for Registry Mirror. Users with a registry mirror and providing a custom CA cert will need to rotate the certificate served by the registry mirror endpoint before using the new EKS-A version. This is true for both new clusters (`create cluster` command) and existing clusters (`upgrade cluster` command).
 - The `--source` option was removed from several package commands. Use either `--kube-version` for registry or `--cluster` for cluster.
 
+## [v0.12.0](https://github.com/aws/eks-anywhere/releases/tag/v0.12.0)
+
+### Added
+- Add support for EKS Anywhere with provider Cloudstack
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:16]:
```suggestion
- Add support for EKS Anywhere with provider CloudStack
```

File: docs/content/en/docs/reference/changelog.md [hunk 6]
```diff
@@ -10,6 +10,27 @@ weight: 35
...
+- Add support to upgrade Bare Metal cluster
+- Add support for using Registry Mirror for Bare Metal
+- RHEL availability for vSphere, Cloudstack and Bare Metal cluster
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:19]:
```suggestion
- RHEL availability for vSphere, CloudStack and Bare Metal cluster
```

File: docs/content/en/docs/reference/changelog.md [hunk 7]
```diff
@@ -10,6 +10,27 @@ weight: 35
...
+- Allow authenticated image pull using Registry Mirror for Ubuntu on vSphere cluster
+- Add option to disable vSphere CSI driver [#3148](https://github.com/aws/eks-anywhere/issues/3148)
+- Add support for skipping load balancer deployment for Bare Metal so users can use their own load balancers [#3608](https://github.com/aws/eks-anywhere/pull/3608)
+- Add support to configure aws-iam-authenticator on workload clusters independent of management cluster [#2814](https://github.com/aws/eks-anywhere/issues/2814)
+- Add EKS Anywhere Packages support for remote management on workload clusters (For more EKS Anywhere packages info: [v0.12.0](https://github.com/aws/eks-anywhere-packages/releases/tag/v0.12.0))
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:24]:
```suggestion
- Add EKS Anywhere Packages support for remote management on workload clusters. (For more EKS Anywhere packages info: [v0.12.0](https://github.com/aws/eks-anywhere-packages/releases/tag/v0.12.0))
```

File: docs/content/en/docs/reference/changelog.md [hunk 8]
```diff
@@ -10,6 +10,27 @@ weight: 35
...
+- Add new EKS Anywhere Packages
+  - AWS Distro for OpenTelemetry (ADOT)
+  - Cert Manager
+  - Cluster Autoscaler
+
+### Fixed
+- Remove special cilium network policy with `policyEnforcementMode` is set to `always` due to lack of pod network connectivity for vsphere csi
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:31]:
```suggestion
- Remove special cilium network policy with `policyEnforcementMode` set to `always` due to lack of pod network connectivity for vSphere CSI
```

File: docs/content/en/docs/reference/changelog.md [hunk 9]
```diff
@@ -10,6 +10,27 @@ weight: 35
...
+- Fixed [#3391](https://github.com/aws/eks-anywhere/issues/3391) [#3560](https://github.com/aws/eks-anywhere/issues/3560) for AWSIamConfig upgrades on EKS-A workload clusters
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:32]:
```suggestion
- Fixed [#3391](https://github.com/aws/eks-anywhere/issues/3391) [#3560](https://github.com/aws/eks-anywhere/issues/3560) for AWSIamConfig upgrades on EKS Anywhere workload clusters
```

**Review by chrisnegus — COMMENTED** [review]
Lots of great features for this release. Just a few small editing suggestions.

File: docs/content/en/docs/reference/changelog.md [hunk 10]
```diff
@@ -10,6 +10,27 @@ weight: 35
...
+- RHEL availability for vSphere, CloudStack and Bare Metal cluster
+- Allow authenticated image pull using Registry Mirror for Ubuntu on vSphere cluster
+- Add option to disable vSphere CSI driver [#3148](https://github.com/aws/eks-anywhere/issues/3148)
+- Add support for skipping load balancer deployment for Bare Metal so users can use their own load balancers [#3608](https://github.com/aws/eks-anywhere/pull/3608)
+- Add support to configure aws-iam-authenticator on workload clusters independent of management cluster [#2814](https://github.com/aws/eks-anywhere/issues/2814)
+- Add EKS Anywhere Packages support for remote management on workload clusters. (For more EKS Anywhere packages info: [v0.12.0](https://github.com/aws/eks-anywhere-packages/releases/tag/v0.12.0))
+- Add new EKS Anywhere Packages
+  - AWS Distro for OpenTelemetry (ADOT)
+  - Cert Manager
+  - Cluster Autoscaler
```
Comment (TerryHowe) [inline docs/content/en/docs/reference/changelog.md:28]:
```suggestion
  - Cluster Autoscaler
  - Metrics Server
```

Comment (panktishah26) [reply docs/content/en/docs/reference/changelog.md:28]: Added

**Review by vivek-koppuru — APPROVED** [review]

File: docs/content/en/docs/reference/changelog.md [hunk 11]
```diff
@@ -10,6 +10,28 @@ weight: 35
 - Certificates signed with SHA-1 are not supported anymore for Registry Mirror. Users with a registry mirror and providing a custom CA cert will need to rotate the certificate served by the registry mirror endpoint before using the new EKS-A version. This is true for both new clusters (`create cluster` command) and existing clusters (`upgrade cluster` command).
 - The `--source` option was removed from several package commands. Use either `--kube-version` for registry or `--cluster` for cluster.
 
+## [v0.12.0](https://github.com/aws/eks-anywhere/releases/tag/v0.12.0)
+
+### Added
+- Add support for EKS Anywhere with provider CloudStack
+- Add support to upgrade Bare Metal cluster
+- Add support for using Registry Mirror for Bare Metal
+- RHEL availability for vSphere, CloudStack and Bare Metal cluster
```
Comment (vignesh-goutham) [inline docs/content/en/docs/reference/changelog.md:19]:
```suggestion
- Redhat-based node image support for vSphere, CloudStack and Bare Metal EKS-A clusters
```

**Pull request merged**